rules/windows/discovery_adfind_command_activity.toml

```diff
@@ -0,0 +1,64 @@
+[metadata]
+creation_date = "2020/10/19"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/10/19"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule detects the Active Directory query tool, AdFind.exe. AdFind has legitimate purposes, but it is frequently
+leveraged by threat actors to perform post-exploitation Active Directory reconnaissance. The AdFind tool has been
+observed in Trickbot, Ryuk, Maze, and FIN6 campaigns. This rule requires Sysmon.
+"""
+index = ["logs-endpoint.events.*", "winlogbeat-*"]
+language = "kuery"
+license = "Elastic License"
+name = "AdFind Command Activity"
+note = "`AdFind.exe` is a legitimate domain query tool. Rule alerts should be investigated to identify if the user has a role that would explain using this tool and that it is being run from an expected directory and endpoint. Leverage the exception workflow in the Kibana Security App or Elasticsearch API to tune this rule to your environment."
+references = [
+    "http://www.joeware.net/freetools/tools/adfind/",
+    "https://thedfirreport.com/2020/05/08/adfind-recon/",
+    "https://www.fireeye.com/blog/threat-research/2020/05/tactics-techniques-procedures-associated-with-maze-ransomware-incidents.html",
+    "https://www.cybereason.com/blog/dropping-anchor-from-a-trickbot-infection-to-the-discovery-of-the-anchor-malware",
+    "https://www.fireeye.com/blog/threat-research/2019/04/pick-six-intercepting-a-fin6-intrusion.html",
+    "https://usa.visa.com/dam/VCOM/global/support-legal/documents/fin6-cybercrime-group-expands-threat-To-ecommerce-merchants.pdf",
+]
+risk_score = 21
+rule_id = "eda499b8-a073-4e35-9733-22ec71f57f3a"
+severity = "low"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Discovery"]
+type = "query"
+
+query = '''
+event.category:process and event.type:start and (process.name:AdFind.exe or process.pe.original_file_name:AdFind.exe)
```

Comment:
@peasead Do we want to be more specific in our targeting of `AdFind.exe` activity?  Are there switches or captured command-lines we may consider higher fidelity than general usage with the tool? 

These [examples](https://thedfirreport.com/2020/05/08/adfind-recon/) may be good starting candidates:
- objectcategory=person – Finds all person objects
- objectcategory=computer – Finds all computers in domain
- trustdmp – Dumps trust objects.
- objectcategory=subnet – Finds all subnets
- domainlist – Dumps all Domain NCs in forest in sorted DNS list format
- dcmodes – Shows modes of all DCs in forest from config
- adinfo – Shows Active Directory Info with whoami info.
- dclist – Dumps Domain Controllers FQDNs.
- computers_pwdnotreqd – Dumps users set with password not required.

Reply:
Thanks Dan.

I actually had thought about making those requirements, but when reading through the [documentation](https://www.joeware.net/freetools/tools/adfind/usage.htm) I felt hyper-focusing on reported usages may miss a creative way an adversary uses `AdFind.exe` and have false negatives. 

I tried to call that out in the `notes` section.

Open to discuss.

Reply:
Good comments. How's this look? I'll need to change it to EQL if this looks right
```
process where event.type == start and 
(process.name : "AdFind.exe" or process.pe.original_file_name == "AdFind.exe") and 
process.args : ("objectcategory=computer", "(objectcategory=computer)", "objectcategory=person", 
"(objectcategory=computer)", "objectcategory=person", "(objectcategory=person)", "objectcategory=subnet", 
"(objectcategory-subnet)", "objectcategory=group", "(objectcategory=group)", 
"objectcategory=organizationalunit", "(objectcategory-organizationalunit), "objectcategory=attributeschema", 
"(objectcategory=attributeschema)"), "domainlist", "dcmodes", "adinfo", "dclist", "computers_pwnotreqd", "trustdmp") 
```

Reply:
Some slight tweaks needed

```sql
process where event.type in ("start", "process_started") and 
  (process.name : "AdFind.exe" or process.pe.original_file_name == "AdFind.exe") and 
  process.args : ("objectcategory=computer", "(objectcategory=computer)", "objectcategory=person", 
                  "(objectcategory=computer)", "objectcategory=person", "(objectcategory=person)", "objectcategory=subnet", 
                  "(objectcategory-subnet)", "objectcategory=group", "(objectcategory=group)", 
                  "objectcategory=organizationalunit", "(objectcategory-organizationalunit)", "objectcategory=attributeschema", 
                  "(objectcategory=attributeschema)", "domainlist", "dcmodes", "adinfo", "dclist", "computers_pwnotreqd", "trustdmp") 
```

Reply:
So close. Fix't and pushed.